Posh-ACME/Plugins/Infomaniak.ps1

```diff
@@ -0,0 +1,256 @@
+            {
+                Write-Verbose "Record $RecordName deleted."
+            }
+        } catch { throw }
+    } else {
+        Write-Debug "Could not find record $RecordName to delete. Nothing to do."
+    }
+
+    <#
+    .SYNOPSIS
+        Remove a DNS TXT record from Infomaniak.
+    .DESCRIPTION
+        Uses the Infomaniak DNS API to remove DNS TXT record.
+    .PARAMETER RecordName
+        The fully qualified name of the TXT record.
+    .PARAMETER TxtValue
+        The value of the TXT record.
+    .PARAMETER InfomaniakToken
+        The API token for your Infomaniak account. This SecureString version can only be used on Windows or any OS running PowerShell 6.2 or later.
+    .PARAMETER InfomaniakTokenInsecure
+        The API token for your Infomaniak account. This standard String version may be used on any OS.
+    .PARAMETER ExtraParams
+        This parameter can be ignored and is only used to prevent errors when splatting with more parameters than this function supports.
+    .EXAMPLE
+        Remove-DnsTxt '_acme-challenge.example.com' 'txt-value' -InfomaniakTokenInsecure 'xxxxxxxx'
+        Removes the specified TXT record with the specified value.
+    #>
+}
+
+function Save-DnsTxt {
+    [CmdletBinding()]
+    param(
+        [Parameter(ValueFromRemainingArguments)]
+        $ExtraParams
+    )
+    <#
+    .SYNOPSIS
+        Not required.
+    .DESCRIPTION
+        This provider does not require calling this function to commit changes to DNS records.
+    .PARAMETER ExtraParams
+        This parameter can be ignored and is only used to prevent errors when splatting with more parameters than this function supports.
+    #>
+}
+
+############################
+# Helper Functions
+############################
+
+# API Docs: https://api.infomaniak.com/doc
+
+Function Find-InfomaniakZone {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory, Position=0)]
+        [string]$RecordName,
+        [Parameter(Mandatory, Position=1)]
+        [hashtable]$RestParameters
+    )
+
+    # setup a module variable to cache the record to zone mapping
+    # so it's quicker to find later
+    if (!$script:InfomaniakRecordZones) { $script:InfomaniakRecordZones = @{} }
+
+    # check for the record in the cache
+    if ($script:InfomaniakRecordZones.ContainsKey($RecordName)) {
+        Write-Debug "Result from Cache $($script:InfomaniakRecordZones.$RecordName.name)"
+        return $script:InfomaniakRecordZones.$RecordName
+    }
+
+    # We need to find the closest/deepest
+    # sub-zone that would hold the record rather than just adding it to the apex. So for something
+    # like _acme-challenge.site1.sub1.sub2.example.com, we'd look for zone matches in the following
+    # order:
+    # - site1.sub1.sub2.example.com
+    # - sub1.sub2.example.com
+    # - sub2.example.com
+    # - example.com
+
+    # remove _acme-challenge. from requested RecordName to get the zone
+    $zone = $RecordName.Split('.',2)[1];
```

Comment:
While this might seem like an easy optimization at first, it can actually break things for people using CNAME aliases for challenges. The record being created may not always be `_acme-challenge.blah`, so you have to check the whole thing first even if most of the time it will fail.

Reply:
Thanks. Will try with an alias.

Reply:
Specifically, try running the following which is what can theoretically happen when people point an alias to the root of a domain:

```powershell
Publish-Challenge _acme-challenge.example.com (Get-PAAccount) faketoken Infomaniak $pArgs -DnsAlias aliasdomain.com -Verbose
```

In this case `aliasdomain.com` is what gets sent to the plugin in the `$RecordName` variable.